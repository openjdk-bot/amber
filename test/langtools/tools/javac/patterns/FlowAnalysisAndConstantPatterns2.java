/*
 * @test /nodynamiccopyright/
 * @summary Stress test flow analysis with constant patterns
 * @compile/fail/ref=FlowAnalysisAndConstantPatterns2.out -XDrawDiagnostics FlowAnalysisAndConstantPatterns2.java
 */
public class FlowAnalysisAndConstantPatterns2 {
    public static int getInt(){
        return 42;
    }
    public static void main(String[] args) {

        int i = getInt();
        switch (i) {
            case 42+1: System.out.println("42+1!");
            case 43: System.out.println("43!");
        }

        switch (i) {
            case 43: System.out.println("43!");
            case 42+1: System.out.println("42+1!");
        }

        switch (i) {
            case 43: System.out.println("43!");
            case 42+1: System.out.println("42+1!");
            case int j: System.out.println("it's an int!");
        }

        switch (i) {
            case 43: System.out.println("43!");
            case int j: System.out.println("it's an int!");
            case 42+1: System.out.println("42+1!");
        }
    }
}
