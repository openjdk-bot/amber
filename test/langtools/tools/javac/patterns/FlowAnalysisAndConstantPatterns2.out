FlowAnalysisAndConstantPatterns2.java:15:13: compiler.err.duplicate.case.label
FlowAnalysisAndConstantPatterns2.java:20:13: compiler.err.duplicate.case.label
FlowAnalysisAndConstantPatterns2.java:25:13: compiler.err.duplicate.case.label
FlowAnalysisAndConstantPatterns2.java:32:13: compiler.err.duplicate.case.label
4 errors